source/base/data_out_base.cc

```diff
@@ -782,6 +791,134 @@ namespace
       }
   }
 
+  /**
+   * Given (i,j,k) coordinates within the Lagrange cell, return an
+   * offset into the local connectivity array
```

Comment:
Say somewhere that this is the 2d version (it is clear from the `order` argument, but not when looking at the code for the first time).